### PR TITLE
Update execution options APIs

### PR DESCRIPTION
**Pull request by waltsims**
Removes changes from user brubbel in case that they do not give concent for relicensing.

### CODE REVIEW

File: kwave/executor.py [hunk 1]
```diff
@@ -32,9 +32,8 @@ def _make_binary_executable(self):
             raise FileNotFoundError(f"Binary not found at {binary_path}")
         binary_path.chmod(binary_path.stat().st_mode | stat.S_IEXEC)
 
-    def run_simulation(self, input_filename: str, output_filename: str, options: str):
```
Comment (waltsims) [inline kwave/executor.py:35]: This removes changes by brubbel

File: kwave/executor.py [hunk 2]
```diff
@@ -32,9 +32,8 @@ def _make_binary_executable(self):
...
-        command = [str(self.execution_options.binary_path), "-i", input_filename, "-o", output_filename]
-        command.extend(options.split(' '))
```
Comment (waltsims) [inline kwave/executor.py:37]: This line was contributed by bubbel and needs to be removed if they do not agree to new license

File: tests/test_simulation_execution_options.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def test_default_initialization(self):
         self.assertEqual(options.kwave_function_name, "kspaceFirstOrder3D")
         self.assertTrue(options.delete_data)
         self.assertIsNone(options.device_num)
-        self.assertEqual(options.num_threads, os.cpu_count())  # "all" should default to CPU count
+        self.assertEqual(options.num_threads, "all")  # "all" should default to CPU count
```
Comment (waltsims) [inline tests/test_simulation_execution_options.py:29]: I need to revisit this

File: tests/test_simulation_execution_options.py [hunk 4]
```diff
@@ -94,75 +99,135 @@ def test_get_options_string_darwin(self):
...
+            f"{os.cpu_count()}",
+            "--verbose",
+            "2",
+            "--p_raw",
+            "--u_max",
+            "-s",
+            f"{self.mock_sensor.record_start_index}"  # Updated to use self.mock_sensor
+        ]
+        for element in expected_elements:
+            self.assertIn(element, options_list)
```
Comment (waltsims) [inline tests/test_simulation_execution_options.py:116]: I should also examine the order but not required

**Comment by waltsims** [comment]
This PR ima to remove code contributed by Brubbel but refactors in the process

File: tests/test_simulation_execution_options.py (hunk 4 above)
Comment (waltsims) [reply tests/test_simulation_execution_options.py]: done

**Comment by faridyagubbayli** [comment]
@waltsims given that the user [has given consent](https://github.com/waltsims/k-wave-python/discussions/525#discussioncomment-11593951), do we still need this PR?

**Comment by waltsims** [comment]
@faridyagubbayli it improves the apis and removes extra spaces from the option strings. These improvements are still useful in my opinion

**Review by faridyagubbayli — APPROVED** [review]
LGTM!

**Pull request merged**